### PR TITLE
feat(clerk-js): Scroll the card into view when opening in UP/OP

### PR DESCRIPTION
**Pull request by octoper**
## Description

This PR introduces some changes that scroll the cards inside UP/OP into view when they are opened.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Comment by octoper** [comment]
!preview

File: packages/clerk-js/src/ui/elements/Action/ActionOpen.tsx [hunk 1]
```diff
@@ -1,16 +1,29 @@
 import type { PropsWithChildren } from 'react';
+import React, { useEffect, useRef } from 'react';
 
 import { useActionContext } from './ActionRoot';
 
 type ActionOpenProps = PropsWithChildren<{ value: string }>;
 
+const ScrollWrapper = React.forwardRef<HTMLDivElement, PropsWithChildren>((props, ref) => {
+  return <div ref={ref}>{props.children}</div>;
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/elements/Action/ActionOpen.tsx:9]: We can do `<div ref={ref} {...props} />` here.

**Comment by octoper** [comment]
!preview

File: packages/clerk-js/src/ui/elements/Action/ActionOpen.tsx [hunk 2]
```diff
@@ -1,16 +1,29 @@
 import type { PropsWithChildren } from 'react';
+import React, { useEffect, useRef } from 'react';
 
 import { useActionContext } from './ActionRoot';
 
 type ActionOpenProps = PropsWithChildren<{ value: string }>;
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/elements/Action/ActionOpen.tsx:6]: Let's move this below the `<ScrollWrapper />` as it relates to  `<Action.Open />`

**Review by desiprisg — COMMENTED** [review]
Are we sure this plays well when there is a field with `autofocus` in the card? Browser might get in the way of the scrolling into view.

File: packages/clerk-js/src/ui/elements/Action/ActionOpen.tsx (hunk 1 above)
Comment (octoper) [reply packages/clerk-js/src/ui/elements/Action/ActionOpen.tsx]: Done!

File: packages/clerk-js/src/ui/elements/Action/ActionOpen.tsx (hunk 2 above)
Comment (octoper) [reply packages/clerk-js/src/ui/elements/Action/ActionOpen.tsx]: Done!

**Comment by octoper** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @octoper, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-it5wjbyc5.clerkstage.dev) | Feb 07, 2024 02:23 PM |

**Comment by octoper** [comment]
!preview

**Review by anagstef — APPROVED** [review]

**Pull request merged**